### PR TITLE
feat(react): add startTransition

### PR DESCRIPTION
**Pull request by r17x**
Reference: https://react.dev/reference/react/startTransition

### CODE REVIEW

**Comment by davesnx** [comment]
Thanks @r17x

File: src/React.re [hunk 1]
```diff
@@ -872,10 +872,15 @@ module Uncurried = {
     "useCallback";
 };
 
+type callback('input, 'output) = 'input => 'output;
```
Comment (anmonteiro) [inline src/React.re:875]: this is already defined in line 471

Comment (anmonteiro) [reply src/React.re:875]: just saw https://github.com/reasonml/reason-react/commit/757dd23e1b1c22a2fc82fd48b4951c65e84235d2

Comment (r17x) [reply src/React.re:875]: Sorry for confusing - This PR should be PASSED Pipelines before merge. @anmonteiro

**Pull request merged**